### DISCUSSION TITLE
Act as Android's gallery app

### DISCUSSION THREAD

**Discussion by TommyTran732** [ente/ente#975] [Enhancements]
The app currently does not act as Android's gallery app. Tapping on pictures in the File Manager (documentsUI) will not work if no other gallery app is installed.

I would be nice if you can have it acting as a gallery app as well.

**Comment by mnvr** [comment]
Right, this is indeed on our roadmap! https://roadmap.ente.io/set-as-default-gallery-app-p-1528/ 

Thank you for the +1. It is unfortunately not a small undertaking, but we we will try to get to this sooner indeed!

**Reply by avocadoattack** [reply]
Hi there! This is a **much-needed** feature, do you have an update on the current status of this, please?

**Reply by noaho** [reply]
For me this is required to onboard my family to using ente, 

If they take photos with the camera and clicking the preview takes them to the wrong gallery, they will organise photos within there and then complain that photos they deleted from the gallery still show in ente.

**Reply by KeSch** [reply]
+1 ...also two and a half years after submitting the feature request ;)

Come on guys please don't let us down on this and give us a status update.

**Reply by vishnukvmd** [reply]
We've started working on this, but don't have a clear ETA yet. The idea is to convert Ente Photos into a local-first version that provides existing features (like machine learning) without having to create an account, and then iterate on that.

**Reply by ch2a** [reply]
> We've started working on this, but don't have a clear ETA yet. The idea is to convert Ente Photos into a local-first version that provides existing features (like machine learning) without having to create an account, and then iterate on that.

Just to be clear, there won't be a dedicated app, right?

**Reply by vishnukvmd** [reply]
No, we will be re-factoring / re-writing parts of the existing app to support this model.

**Reply by devasworski** [reply]
A big +1 from my side as well.
It's a required feature for me and I am really looking forward to it.

**Reply by nicpaesk** [reply]
Any news ? Same for me in terms of onboarding family and friends, they get super confused

**Reply by laurenspriem** [reply]
The local gallery version of Ente Photos is there on mobile, but there's still more work needed to make it a full gallery app. 

@nicpaesk could you explain where the main confusion with them was, so that we can take it into account as we continue to work on it?

**Reply by neonspectra** [reply]
@laurenspriem Not that person, but FWIW, most non-technical people do not understand the difference between the gallery app and a separate photos app. They just assume that any app with their photos will just do all the things. They don't think about what files go where or what apps contain what information because the concept of files is starting to elude people.

The biggest experience I have had with this is that people get confused when open an "upload image" interface and it shows a different set of photos than what's in Ente. This is of course because Ente does not act as a full gallery app, and I know this sounds ridiculous to people like us but explaining this to someone with zero technical knowledge makes their eyes glaze over. I think Ente is really great, but telling someone that to send a photo from Ente in a different app they have to open a separate app -> save image from that app to their gallery app (or copy image to clipboard, except this doesn't work with videos) -> use their phone gallery to upload images or whatnot goes over peoples' heads and is too annoying for them to be willing to learn to do.

**Reply by nicpaesk** [reply]
> @nicpaesk could you explain where the main confusion with them was, so that we can take it into account as we continue to work on it?

Yes @laurenspriem here are a few examples: 

- On whatsapp, when they need to select a picture, only the pictures in the gallery appear, not Ente ones. Same when selecting a photo to upload to chatgpt app, or any other app. As such, they have to exit the app, enter in Ente, and from there share pictures, or save locally before sharing. It creates a lot of friction, and they don't even remember what's the difference between Ente and their gallery.  

- When taking photos on the camera app, when they click on the photos miniature, they are redirected to the gallery app not ente

- When setting up ente, they need to manually select which local albums should be imported / backed up (for instance the whatsapp ones). However, they need to go to the configuration to set this up, it doesn't propose on it's own. So I had to do this for them. 

They didn't have these problems when they were in google photos. Now I moved them to GrapheneOS and Ente but they are quite reluctant.

**Reply by Hikiru** [reply]
One huge issue for me is that I can't open Ente from my camera to look at the last photo I took. If I use another gallery app for this, then deleting the photo I just took won't delete it from Ente, since it doesn't delete photos deleted locally (a big annoyance of mine that I wish would be changed).

**Comment by laurenspriem** [comment]
We've shipped a significant improvement in our last v1.3.44 release! For supported apps and devices, it's now possible to:
1. Pick photos and videos from Ente in other apps
2. Open camera shots directly in Ente

You can find more info [here in our docs](https://ente.com/help/photos/features/utilities/android-gallery)!

More improvements towards a full-fledged coming soon!

**Reply by laurenspriem** [reply]
Side note: for people using [Gcam Services Provider](https://github.com/lukaspieper/Gcam-Services-Provider) with GrapheneOS the camera view currently opens the home gallery. On the next release (~2 weeks) it will also go to the photo viewer of the most recent photo directly.

**Reply by zilexa** [reply]
> We've shipped a significant improvement in our last v1.3.44 release! For supported apps and devices, it's now possible to:
> 
>     1. Pick photos and videos from Ente in other apps
> 
>     2. Open camera shots directly in Ente
> 
> 
> You can find more info [here in our docs](https://ente.com/help/photos/features/utilities/android-gallery)!
> 
> More improvements towards a full-fledged coming soon!

Ente is not shown as one of the available apps when i want to select a photo in another app, on a Pixel 9 Pro..Other apps are shown, in my case the Google Photos app, Koofr, CX File Explorer. 
It's not a big deal though.. really not a priority to get this working versus the more serious basic stuff.. like common workflows that destroy metadata (rotating for example).

**Reply by laurenspriem** [reply]
@zilexa could you check if it's still not showing in the latest [v1.3.56](https://github.com/ente-io/ente/releases/tag/photos-v1.3.56) release? There was some regression but it should be fixed now.

As for the editing worksflows you're talking about, I want to clarify that we're not destroying any metadata. Our current approach is that edits create a copy, and the original stays untouched and keeps all metadata intact. But I agree there should be a better approach where we don't make a copy but keep both the metadata and edits in a single file. This is on our roadmap to implement soon!

**Reply by zilexa** [reply]
Workflows do absolutely destroy data. I specifically use the term workflow: When a noob rotates a photo in Ente, the user will then be presented a "good' version and a "bad" version, the noob deletes the bad version, because the noob only wants the rotated version. Duh. 
Thats what I mean with a destructive *workflow*. Constantly claiming that Ente is non-destructive, while basic actual *use* cases lead to loss of data is not good.
Plenty of people, paying subscribers, have trusted Ente but ran into this problem already. 
And this is not being solved. What we get are social updates mostly (although I'm sure there are plenty of backend updates that lead to better performance), because someone told Ente Product Team that those features will attract more subscribers (as written in an interview).
Note that Ente does not not even warn or inform the user during the workflow that this is going to lead to loss of metadata.

I tested: Ente does show as one of the apps now, when using the files picker. 
Some apps also allow to pick photos specifically. This then shows the (pretty useless) Android Cloud Media picker, which only shows Google Photos. But I guess thats really a Google only thing, Google trying to push their Photos app.

**Reply by hedgefield** [reply]
> @zilexa I agree there should be a better approach where we don't make a copy but keep both the metadata and edits in a single file. This is on our roadmap to implement soon!

Very excited to hear there are upcoming improvements in the editing workflows, looking forward to it!

Just wanted to chime in and say I hope it will eventually also become possible to open camera shots in ente on oxygenOS/colorOS, so I can ditch Oneplus' photos app entirely.

Keep up the good work 🙏

**Comment by dm30** [comment]
Hey guys. I've been playing with Gallery Mode which, in my case, is most useful as an offline mode for showing Ente to those more reluctant to move to yet another, unknown service. Those who don't want to create another account, suspicious of receiving spam, unwanted notifications, you get it.
I found it pleasant that widget for 'memories' actually works (the doc. seemed to indicate the opposite to me). Still, I think some improvements could be done:
- Face recognition and clustering is there, but you can't put a label on them? A local DB should be built. This would unlock the people widget. I found it awkward having all my loved ones conveniently organized there not having the option.
- Selecting favorites is a must, even if that means creating the corresponding album, which leads to:
- The ability to create local albums. Maybe creating corresponding folders to keep things tidy. That should be achievable in Android right? And would unlock Album widgets.
Perhaps not as straightforward on iOS, though, but these changes would make for good showcase of Ente's range of offline capabilities to the more reticent out there.
I'm soon going to try convincing some relatives to come to the platform and maybe it will be easier trying this route with some.
I hope it helps. Thank you so much.
